### PR TITLE
Teleportation environment

### PR DESCRIPTION
**Pull request by jwallnoefer**
This pull request adds an environment in which the agent is expected to learn the quantum teleportation protocol.

### TODO
- [x]  Get tests to pass.
- [x]  Write README
- [x]  Wait until testing branch is merged with master.

------------------
Before merging I will need to expand the README to better discuss the purpose and limitations of the environment. But for now I would kindly ask you to review whether I adapted my environment correctly to match your standard.

### CODE REVIEW

**Comment by HendrikPN** [comment]
Thanks a lot for the environment! We are very happy for it. I will have a closer look soon, but at first glance all looks fine. We are about to include unit tests to scigym. If you want, you can run you environment against the tests. You can find the current tests on this [branch](https://github.com/HendrikPN/scigym/tree/issue_3_testing)

UPDATE: Currently, however it seems our tests run into problems with your environment, but that might just be an error on our side. The error is,
```
TypeError: '<=' not supported between instances of 'int' and 'tuple'
```
which seems to be cause by your `MultiDiscrete` observation space.

**Comment by jwallnoefer** [comment]
I managed to get the tests to pass. It was indeed me violating some scigym standards!

I'm not sure if merging this branch with the testing branch was the best way to run the tests on my environment. - Please let me know if there's a better way that doesn't look so confusing.

On second thought... I think with the merge I somehow destroyed the github history of the branches... oh well...

**Comment by HendrikPN** [comment]
Yeh. Maybe wait for the testing branch to be merged to master before merging the master into your branch.

**Comment by jwallnoefer** [comment]
Ok... I'll do that!

**Comment by HendrikPN** [comment]
#6 is resolved, you can include the testing from the master now!

File: setup.py
Comment (HendrikPN) [inline setup.py]: You don't need to specify `numpy`. It is assumed to be available.

File: scigym/envs/quantum_physics/quantum_computing/teleportation/README.md
Comment (HendrikPN) [inline scigym/envs/quantum_physics/quantum_computing/teleportation/README.md:76]: I suppose you will improve on this before merging!

**Comment by jwallnoefer** [comment]
The environment is finished from my side. Please let me know if anything needs changing or if anything is unclear from the Documentation.

File: scigym/envs/quantum_physics/quantum_computing/teleportation/README.md
Comment (HendrikPN) [inline scigym/envs/quantum_physics/quantum_computing/teleportation/README.md:76]: Do you not want to add your paper reference somewhere?

**Review by HendrikPN — APPROVED** [review]

File: scigym/envs/quantum_physics/quantum_computing/teleportation/README.md
Comment (jwallnoefer) [reply scigym/envs/quantum_physics/quantum_computing/teleportation/README.md:76]: That's probably a good idea.

**Review by HendrikPN — APPROVED** [review]

**Pull request merged**
